### PR TITLE
Upgrade to ES6 syntax, nock test

### PR DESCRIPTION
**Pull request by Pchelolo**
1. Drop support for node < 6
2. Eslint
3. Nock for tests
4. ES6 syntax
Next steps: mode code coverage, get rid of custom retry functionality

cc @wikimedia/services

### CODE REVIEW

**Comment by Pchelolo** [comment]
I have 2 more based on this one

File: package.json [hunk 1]
```diff
@@ -1,20 +1,34 @@
 {
   "name": "preq",
-  "version": "0.5.6",
+  "version": "0.6.0",
```
Comment (d00rman) [inline package.json]: No functional change, so we can bump the version in a PR that brings new functionality

**Review by d00rman — CHANGES_REQUESTED** [review]
LGTM, one minor about the version.

File: package.json (hunk 1 above)
Comment (Pchelolo) [reply package.json]: Done @d00rman

**Pull request merged**